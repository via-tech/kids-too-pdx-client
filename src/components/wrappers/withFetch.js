import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { SyncLoader } from 'react-spinners';
import styles from './Wrappers.css';

export const withFetch = Component => {
  class WithFetch extends PureComponent {
    static propTypes = {
      fetch: PropTypes.func.isRequired,
      loading: PropTypes.bool.isRequired
    };

    static defaultProps = {
      loading: false
    };

    componentDidMount() {
      this.props.fetch();
    }

    render() {
      const { loading } = this.props;
      return (
        <div className={styles.Wrappers}>
          {loading ?
            <div className={styles.Loader}>
              <SyncLoader 
                sizeUnit={'px'}
                size={30}
                color={'#123abc'}
              />
            </div>
            : <Component {...this.props} />}
        </div>
      );
    }
  }
  return WithFetch;
};
