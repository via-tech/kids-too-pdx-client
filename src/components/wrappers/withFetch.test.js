import React from 'react';
import renderer from 'react-test-renderer';
import { withFetch } from './withFetch';

describe('With Fetch wrapper', () => {
  it('snapshots withFetch', () => {
    const props = {
      fetch: jest.fn(),
      loader: true
    };

    const Component = withFetch(() =>
      <div>
        <h1>The Main Component</h1>
      </div>
    );

    const tree = renderer.create(
      <Component {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
