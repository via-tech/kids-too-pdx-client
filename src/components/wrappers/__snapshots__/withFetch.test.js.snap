// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Fetch wrapper snapshots withFetch 1`] = `
<div
  className="Wrappers"
>
  <div>
    <h1>
      The Main Component
    </h1>
  </div>
</div>
`;
